server/upstream/download-pypi-versions

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env python
+
+#
+# Copyright (c) 2014, SUSE LINUX Products GmbH
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+# USA
+#
+# (Licensed under the LGPLv2.1 or later)
+#
+#
+# Authors: Thomas Bechtold <tbechtold@suse.com>
+#
+
+import argparse
+import functools
+import os
+import sys
+import time
+from multiprocessing import Pool
+
+try:
+    import xmlrpclib
+except ImportError:
+    # Python3 support
+    import xmlrpc.client as xmlrpclib
+
+
+XMLRPC_SERVER_PROXY = 'https://pypi.python.org/pypi'
+
+
+@functools.total_ordering
+class PackageInfo(object):
+    """Represents a package info"""
+    def __init__(self, type_, name, version, url):
+        self.type = type_.rstrip()
+        self.name = name.rstrip()  # the original name from pypi (without python3- or python- prefix)
+        self.version = version.rstrip()
+        self.url = url.rstrip()
```

Comment:
> Why rstrip and not just strip?

Done in https://github.com/toabctl/osc-plugin-collab/commit/2f370e3bf6fce74afc85f5de1f38382d091f9108